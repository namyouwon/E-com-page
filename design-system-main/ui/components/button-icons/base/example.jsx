// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import ButtonIcon, { HintParent } from '../';
import {Button} from '../../buttons/base/example';
import SvgIcon from '../../../shared/svg-icon';

export default [
  {
    id: 'default',
    label: 'Base â€“ default',
    element: <ButtonIcon assistiveText="Settings" title="Settings" />
  }
];

export let states = [
  {
    id: 'error',
    label: 'Base - Error',
    element: (
      <ButtonIcon
        feedback="error"
        symbol="error"
        title="Error"
        assistiveText="Error"
      />
    )
  },
  {
    id: 'warning',
    label: 'Base - Warning',
    element: (
      <ButtonIcon
        feedback="warning"
        symbol="warning"
        title="Warning"
        assistiveText="Warning"
      />
    )
  },
  {
    id: 'bare-disabled',
    label: 'Base - Disabled',
    element: <ButtonIcon assistiveText="Settings" disabled title="Settings" />
  }
];

export let examples = [
  {
    id: 'hint-hover',
    label: 'Base - Hint on hover',
    element: (
      <HintParent>
        <ButtonIcon
          assistiveText="Settings"
          iconClassName="slds-button__icon_hint"
          title="Settings"
        />
      </HintParent>
    )
  },
  {
    // this was put in to place to test a styling hooks edge case
    id: 'button-icon-lbc-mismatch',
    label: 'Button Icon - LBC Mismatch',
    element: (
      <Button className="slds-button_icon slds-button_icon-border">
        <span className="slds-assistive-text">Error</span>
        <SvgIcon
          sprite="utility"
          symbol="clock"
          className="slds-button__icon slds-icon-text-error"
        />
      </Button>
    )
  }
];
