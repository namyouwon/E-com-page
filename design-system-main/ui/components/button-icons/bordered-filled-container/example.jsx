// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import ButtonIcon, { HintParent } from '../';

const exampleDemoStyles = 'background-color: #f4f6f9; padding: 0.5rem;';

export default [
  {
    id: 'default',
    label: 'Bordered Filled â€“ default',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <ButtonIcon
        assistiveText="Search"
        symbol="search"
        theme="neutral"
        title="Search"
      />
    )
  }
];

export let states = [
  {
    id: 'disabled',
    label: 'Bordered Filled - Disabled',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <ButtonIcon
        assistiveText="Search"
        disabled
        symbol="search"
        theme="neutral"
        title="Search"
      />
    )
  }
];

export let examples = [
  {
    id: 'with-dropdown',
    label: 'Bordered Filled - With dropdown',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <ButtonIcon
        assistiveText="More options"
        hasDropdown
        theme="neutral"
        title="More Options"
      />
    )
  },
  {
    id: 'hint-hover',
    label: 'Bordered Filled - Hint on hover',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <HintParent>
        <ButtonIcon
          assistiveText="More options"
          iconClassName="slds-button__icon_hint"
          theme="neutral"
          title="More Options"
        />
      </HintParent>
    )
  }
];
