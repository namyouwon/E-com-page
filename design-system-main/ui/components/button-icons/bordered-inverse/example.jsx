// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import ButtonIcon, { HintParent } from '../';

export const Context = props => (
  <div style={{ padding: '0.5rem', backgroundColor: '#16325C' }}>
    {props.children}
  </div>
);

const exampleDemoStyles = 'background-color: #16325C; padding: 0.5rem;';

export default [
  {
    id: 'default',
    label: 'Bordered Inverse – default',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <ButtonIcon
        assistiveText="Search"
        size="medium"
        symbol="search"
        theme="inverse"
        title="Search"
      />
    )
  }
];

export let states = [
  {
    id: 'disabled',
    label: 'Bordered Inverse – Disabled',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <ButtonIcon
        assistiveText="Search"
        disabled
        size="medium"
        symbol="search"
        theme="inverse"
        title="Search"
      />
    )
  }
];

export let examples = [
  {
    id: 'with-dropdown',
    label: 'Bordered Inverse – With dropdown',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <ButtonIcon
        assistiveText="More options"
        hasDropdown
        theme="neutral"
        title="More Options"
      />
    )
  },
  {
    id: 'hint-hover',
    label: 'Bordered Inverse – Hint on hover',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <HintParent>
        <ButtonIcon
          assistiveText="More options"
          iconClassName="slds-button__icon_inverse-hint"
          size="medium"
          theme="inverse"
          title="More Options"
        />
      </HintParent>
    )
  }
];
