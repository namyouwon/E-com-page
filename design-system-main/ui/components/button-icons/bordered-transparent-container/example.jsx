// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import ButtonIcon from '../';

const exampleDemoStyles = 'background-color: #f4f6f9; padding: 0.5rem;';

export default [
  {
    id: 'default',
    label: 'Bordered Transparent - default',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <ButtonIcon
        assistiveText="Search"
        title="Search"
        theme="transparent"
        symbol="search"
      />
    )
  }
];

export let states = [
  {
    id: 'disabled',
    label: 'Bordered Transparent - Disabled',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <ButtonIcon
        assistiveText="Search"
        title="Search"
        theme="transparent"
        symbol="search"
        disabled
      />
    )
  }
];

export let examples = [
  {
    id: 'hint-hover',
    label: 'Bordered Transparent - Hint on hover',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <div className="slds-hint-parent">
        <ButtonIcon
          theme="transparent"
          iconClassName="slds-button__icon_hint"
          assistiveText="More options"
          title="More Options"
        />
      </div>
    )
  }
];
