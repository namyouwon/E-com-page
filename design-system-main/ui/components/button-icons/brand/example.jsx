// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import ButtonIcon from '../';

export default [
  {
    id: 'default',
    label: 'Brand – default',
    element: (
      <ButtonIcon
        assistiveText="Search"
        title="Search"
        theme="brand"
        symbol="search"
      />
    )
  }
];

export let states = [
  {
    id: 'disabled',
    label: 'Brand – Disabled',
    element: (
      <ButtonIcon
        assistiveText="Search"
        title="Search"
        theme="brand"
        symbol="search"
        disabled
      />
    )
  }
];
