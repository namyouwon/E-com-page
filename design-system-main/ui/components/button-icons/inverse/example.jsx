// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import ButtonIcon from '../';

const exampleDemoStyles = 'background-color: #16325C; padding: 0.5rem;';

export default [
  {
    id: 'default',
    label: 'Inverse â€“ default',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <ButtonIcon
        assistiveText="Search"
        title="Search"
        theme="inverse"
        symbol="search"
      />
    )
  }
];

export let states = [
  {
    id: 'disabled',
    label: 'Inverse - Disabled',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <ButtonIcon
        assistiveText="Search"
        title="Search"
        theme="inverse"
        symbol="search"
        disabled
      />
    )
  }
];

export let examples = [
  {
    id: 'with-dropdown',
    label: 'Inverse - With dropdown',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <ButtonIcon
        hasDropdown
        theme="transparent"
        className="slds-button_icon-inverse"
        assistiveText="More options"
        title="More Options"
      />
    )
  },
  {
    id: 'hint-hover',
    label: 'Inverse - Hint on hover',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <div className="slds-hint-parent">
        <ButtonIcon
          theme="inverse"
          iconClassName="slds-button__icon_inverse-hint"
          assistiveText="More options"
          title="More Options"
        />
      </div>
    )
  }
];
