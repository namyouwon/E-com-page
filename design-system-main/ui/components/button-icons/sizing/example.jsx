// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import ButtonIcon from '../';

export default [
  {
    id: 'default',
    label: 'Bare â€“ Default',
    element: (
      <ButtonIcon
        assistiveText="More options"
        title="More options"
      />
    )
  }
];

export let examples = [
  {
    id: 'button-icon-large',
    label: 'Bare - Size Large',
    element: (
      <ButtonIcon
        iconClassName="slds-button__icon_large"
        assistiveText="More options"
        title="More options"
      />
    )
  },
  {
    id: 'button-icon-small',
    label: 'Bare - Size Small',
    element: (
      <ButtonIcon
        iconClassName="slds-button__icon_small"
        assistiveText="More options"
        title="More options"
        />
    )
  },
  {
    id: 'button-icon-x-small',
    label: 'Bare - Size Extra Small',
    element: (
      <ButtonIcon
        iconClassName="slds-button__icon_x-small"
        assistiveText="More options"
        title="More options"
      />
    )
  },
  {
    id: 'button-icon-contained-large',
    label: 'Contained - Size Large',
    element: (
      <ButtonIcon
        className="slds-button_icon-border"
        size="large"
        assistiveText="More options"
        title="More options"
      />
    )
  },
  {
    id: 'button-icon-contained-small',
    label: 'Contained - Size Small',
    element: (
      <ButtonIcon
        className="slds-button_icon-border"
        size="small"
        assistiveText="More options"
        title="More options"
      />
    )
  },
  {
    id: 'button-icon-contained-x-small',
    label: 'Contained - Size Extra Small',
    element: (
      <ButtonIcon
        className="slds-button_icon-border"
        size="x-small"
        assistiveText="More options"
        title="More options"
      />
    )
  },
  {
    id: 'button-icon-contained-xx-small',
    label: 'Contained - Size Extra Extra Small',
    element: (
      <ButtonIcon
        className="slds-button_icon-border"
        size="xx-small"
        assistiveText="More options"
        title="More options"
      />
    )
  }
  ];
