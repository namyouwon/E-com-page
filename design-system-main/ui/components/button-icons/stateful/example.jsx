// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import PropTypes from 'prop-types';

import ButtonIcon from '../';

export const ButtonIconStateful = props => (
  <ButtonIcon
    aria-pressed={props['aria-pressed']}
    assistiveText={props.assistiveText}
    disabled={props.disabled}
    selected={props.selected}
    symbol={props.symbol}
    theme={props.theme}
    title={props.title}
  />
);

ButtonIconStateful.propTypes = {
  title: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  assistiveText: PropTypes.string.isRequired,
  'aria-pressed': PropTypes.string,
  disabled: PropTypes.bool,
  selected: PropTypes.bool,
  theme: PropTypes.string
};

ButtonIconStateful.defaultProps = {
  'aria-pressed': false,
  disabled: false,
  selected: false,
  theme: null
};

/**
 * Default
 */

export default [
  {
    id: 'default',
    label: 'Stateful â€“ default',
    element: (
      <ButtonIconStateful
        aria-pressed="false"
        assistiveText="Like"
        symbol="like"
        theme="neutral"
        title="Like"
      />
    )
  }
];

/**
 * States
 */
export const states = [
  {
    id: 'button-icon-stateful-selected',
    label: 'Stateful - Selected',
    element: (
      <ButtonIconStateful
        title="Like"
        symbol="like"
        theme="neutral"
        assistiveText="Like"
        aria-pressed="true"
        selected
      />
    )
  },
  {
    id: 'button-icon-stateful-disabled',
    label: 'Stateful - Disabled',
    element: (
      <ButtonIconStateful
        title="Pin"
        symbol="pin"
        theme="neutral"
        assistiveText="Pin"
        aria-pressed="false"
        disabled
      />
    )
  },
  {
    id: 'button-icon-stateful-selected-disabled',
    label: 'Stateful - Selected Disabled',
    element: (
      <ButtonIconStateful
        title="Pin"
        symbol="pinned"
        theme="neutral"
        assistiveText="Pin"
        aria-pressed="true"
        selected
        disabled
      />
    )
  }
];
