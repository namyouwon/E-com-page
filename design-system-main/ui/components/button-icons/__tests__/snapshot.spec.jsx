/* eslint-env jest */
import React from 'react';
import createHelpers from '../../../../jest.helpers';

import ButtonIcon from '../';
import { ButtonIconStateful } from '../stateful/example';

const { matchesMarkup } = createHelpers(__dirname);

// Bare
describe('bare button icon', () => {
  it('renders a button icon', () => matchesMarkup(<ButtonIcon />));

  it('renders a disabled button icon', () =>
    matchesMarkup(<ButtonIcon disabled />));
});

// Container
describe('button icon with container', () => {
  it('renders a button icon', () =>
    matchesMarkup(<ButtonIcon size="medium" />));

  it('renders a disabled button icon', () =>
    matchesMarkup(<ButtonIcon disabled size="medium" />));
});

// Bordered Container
describe('button icon with bordered transparent container', () => {
  it('renders a button icon', () =>
    matchesMarkup(<ButtonIcon theme="transparent" />));

  it('renders a disabled button icon', () =>
    matchesMarkup(<ButtonIcon disabled theme="transparent" />));
});

// Bordered Container Filled
describe('button icon with bordered filled container', () => {
  it('renders a button icon', () =>
    matchesMarkup(<ButtonIcon theme="neutral" />));

  it('renders a disabled button icon', () =>
    matchesMarkup(<ButtonIcon disabled theme="neutral" />));
});

// Brand
describe('brand button icon', () => {
  it('renders a button icon', () =>
    matchesMarkup(<ButtonIcon theme="brand" />));

  it('renders a disabled button icon', () =>
    matchesMarkup(<ButtonIcon disabled theme="brand" />));
});

// Inverse
describe('button icon on inversed background', () => {
  it('renders a button icon', () =>
    matchesMarkup(<ButtonIcon theme="inverse" />));

  it('renders a disabled button icon', () =>
    matchesMarkup(<ButtonIcon disabled theme="inverse" />));
});

// Dropdown
describe('button menu icon for dropdowns', () => {
  it('renders a button menu icon', () =>
    matchesMarkup(<ButtonIcon hasDropdown theme="neutral" />));

  it('renders a button menu icon on inversed background', () =>
    matchesMarkup(
      <ButtonIcon
        className="slds-button_icon-inverse"
        hasDropdown
        theme="transparent"
      />
    ));
});

// Stateful
describe('stateful button icon', () => {
  it('renders the stateful button icon', () =>
    matchesMarkup(
      <ButtonIconStateful
        aria-pressed="false"
        assistiveText="Like"
        symbol="like"
        theme="neutral"
        title="Like"
      />
    ));

  it('renders a selected stateful button icon', () => {
    matchesMarkup(
      <ButtonIconStateful
        title="Like"
        symbol="like"
        theme="neutral"
        assistiveText="Like"
        aria-pressed="true"
        selected
      />
    );
  });

  it('renders a disabled stateful button icon', () => {
    matchesMarkup(
      <ButtonIconStateful
        title="Pin"
        symbol="pin"
        theme="neutral"
        assistiveText="Pin"
        aria-pressed="true"
        disabled
      />
    );
  });

  it('renders a selected disabled stateful button icon', () => {
    matchesMarkup(
      <ButtonIconStateful
        title="Pin"
        symbol="pinned"
        theme="neutral"
        assistiveText="Pin"
        aria-pressed="true"
        selected
        disabled
      />
    );
  });
});
