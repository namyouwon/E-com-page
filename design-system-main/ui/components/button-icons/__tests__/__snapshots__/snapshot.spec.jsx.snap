// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bare button icon renders a button icon 1`] = `
<button
  className="slds-button slds-button_icon"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`bare button icon renders a disabled button icon 1`] = `
<button
  className="slds-button slds-button_icon"
  disabled={true}
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`brand button icon renders a button icon 1`] = `
<button
  className="slds-button slds-button_icon slds-button_icon-brand"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`brand button icon renders a disabled button icon 1`] = `
<button
  className="slds-button slds-button_icon slds-button_icon-brand"
  disabled={true}
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon on inversed background renders a button icon 1`] = `
<button
  className="slds-button slds-button_icon slds-button_icon-inverse"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon on inversed background renders a disabled button icon 1`] = `
<button
  className="slds-button slds-button_icon slds-button_icon-inverse"
  disabled={true}
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with bordered filled container renders a button icon 1`] = `
<button
  className="slds-button slds-button_icon slds-button_icon-border-filled"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with bordered filled container renders a disabled button icon 1`] = `
<button
  className="slds-button slds-button_icon slds-button_icon-border-filled"
  disabled={true}
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with bordered transparent container renders a button icon 1`] = `
<button
  className="slds-button slds-button_icon slds-button_icon-border"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with bordered transparent container renders a disabled button icon 1`] = `
<button
  className="slds-button slds-button_icon slds-button_icon-border"
  disabled={true}
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with container renders a button icon 1`] = `
<button
  className="slds-button slds-button_icon slds-button_icon-container"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with container renders a disabled button icon 1`] = `
<button
  className="slds-button slds-button_icon slds-button_icon-container"
  disabled={true}
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button menu icon for dropdowns renders a button menu icon 1`] = `
<button
  aria-expanded="false"
  aria-haspopup="true"
  className="slds-button slds-button_icon slds-button_icon-more"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="slds-button__icon slds-button__icon_x-small"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button menu icon for dropdowns renders a button menu icon on inversed background 1`] = `
<button
  aria-expanded="false"
  aria-haspopup="true"
  className="slds-button slds-button_icon slds-button_icon-container-more slds-button_icon-inverse"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="slds-button__icon slds-button__icon_x-small"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`stateful button icon renders a disabled stateful button icon 1`] = `
<button
  aria-pressed="true"
  className="slds-button slds-button_icon slds-button_icon-border-filled"
  disabled={true}
  onClick={[Function]}
  title="Pin"
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#pin"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Pin
  </span>
</button>
`;

exports[`stateful button icon renders a selected disabled stateful button icon 1`] = `
<button
  aria-pressed="true"
  className="slds-button slds-button_icon slds-is-selected slds-button_icon-border-filled"
  disabled={true}
  onClick={[Function]}
  title="Pin"
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#pinned"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Pin
  </span>
</button>
`;

exports[`stateful button icon renders a selected stateful button icon 1`] = `
<button
  aria-pressed="true"
  className="slds-button slds-button_icon slds-is-selected slds-button_icon-border-filled"
  disabled={false}
  onClick={[Function]}
  title="Like"
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#like"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Like
  </span>
</button>
`;

exports[`stateful button icon renders the stateful button icon 1`] = `
<button
  aria-pressed="false"
  className="slds-button slds-button_icon slds-button_icon-border-filled"
  disabled={false}
  onClick={[Function]}
  title="Like"
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#like"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Like
  </span>
</button>
`;
